tensorflow/core/platform/cus.h

```diff
@@ -31,143 +31,90 @@ limitations under the License.
+  CUSTOM_DEVICE_FUNC constexpr cus() : value(0) {}
+  CUSTOM_DEVICE_FUNC cus(const float& f) : value(castF32ToValue(f)) {}
+  CUSTOM_DEVICE_FUNC cus(const double& d) : cus(static_cast<float>(d)) {}
+  explicit CUSTOM_DEVICE_FUNC cus(const complex64& c64)
+      : cus(static_cast<float>(c64.real())) {}
+  explicit CUSTOM_DEVICE_FUNC cus(const complex128& c128)
+      : cus(static_cast<float>(c128.real())) {}
 
   template <class T>
-  explicit constexpr CUSTOM_DEVICE_FUNC cus(const T& value)
-      : cus(static_cast<cusWrappedType>(value)) {}
+  explicit CUSTOM_DEVICE_FUNC cus(const T& value)
+      : cus(static_cast<float>(value)) {}
 
-  CUSTOM_DEVICE_FUNC operator float() const {
-    return static_cast<float>(value);
-  }
+  CUSTOM_DEVICE_FUNC operator float() const { return castValueToF32(value); }
 
   explicit CUSTOM_DEVICE_FUNC operator double() const {
-    return static_cast<double>(value);
+    float f = static_cast<float>(*this);
+    return static_cast<double>(f);
   }
-
-  explicit CUSTOM_DEVICE_FUNC operator uint32_t() const {
-    return static_cast<uint32_t>(value);
-  }
-
-  // template <class T>
-  // explicit CUSTOM_DEVICE_FUNC operator T() const {
-  //   return static_cast<T>(value);
-  // }
-
-  // explicit CUSTOM_DEVICE_FUNC operator cusWrappedType() const { return value;
-  // }
-
-  // CUSTOM_DEVICE_FUNC inline cus& operator=(cusWrappedType i) {
-
-  //   return *this;
-  // }
-
-  // CUSTOM_DEVICE_FUNC inline cus& operator=(const cus& a) {
-  //   this->setValue(static_cast<cusWrappedType>(a));
-  //   return *this;
-  // }
+  static uint32_t castF32ToValue(const float& f);
+  static float castValueToF32(const uint32_t& u);
 };
 
-CUSTOM_DEVICE_FUNC inline cus operator+(const cus& a, const cus& b) {
-  return cus(static_cast<cusWrappedType>(a) + static_cast<cusWrappedType>(b));
-}
-
-CUSTOM_DEVICE_FUNC inline cus operator-(const cus& a) {
-  return cus(-static_cast<cusWrappedType>(a));
-}
-
-CUSTOM_DEVICE_FUNC inline cus operator-(const cus& a, const cus& b) {
-  return cus(static_cast<cusWrappedType>(a) - static_cast<cusWrappedType>(b));
-}
-
-CUSTOM_DEVICE_FUNC inline cus operator*(const cus& a, const cus& b) {
-  return cus(static_cast<cusWrappedType>(a) * static_cast<cusWrappedType>(b));
-}
-
-CUSTOM_DEVICE_FUNC inline cus operator/(const cus& a, const cus& b) {
-  return cus(static_cast<cusWrappedType>(a) / static_cast<cusWrappedType>(b));
-}
 
-CUSTOM_DEVICE_FUNC inline cus operator+=(cus& a, const cus& b) {
+inline float CastCusToF32(cus c) { return (float)(c); }
+inline cus CastF32ToCus(const float f) { return cus(f); }
+cus CusAdd(cus a, cus b);
+cus CusSub(cus a, cus b);
+cus CusMul(cus a, cus b);
+cus CusDiv(cus a, cus b);
+cus CusNeg(cus a);
+
+bool CusEq(cus a, cus b);
+bool CusNe(cus a, cus b);
+bool CusLt(cus a, cus b);
+bool CusLe(cus a, cus b);
+bool CusGt(cus a, cus b);
+bool CusGe(cus a, tensorflow::cus b);
+
+
+inline CUSTOM_DEVICE_FUNC cus operator+(const cus& a, const cus& b){ return CusAdd(a,b);}
```

Comment:
is it not throwing any compiler error? since you are passing const variables to a function with non-const arguments?